### PR TITLE
iOS plugin

### PR DESCRIPTION
**Pull request by CherryPerry**
- Move iOS configuration to plugin
- Remove TODO to turn on/off `check` task dependency on `iosTask`
- Make `check` depends on `iosTask` and all iOS binaries compilation tasks to verify, that lib can be built for all platforms and to simplify Travis configuration (fixes https://github.com/badoo/Reaktive/issues/81)
- Use new API to get path to executable iOS test binary (fixes https://github.com/badoo/Reaktive/issues/89)

### CODE REVIEW

**Comment by arkivanov** [comment]
For me it's fine) @lukaville please review as well

File: buildSrc/src/main/kotlin/IosPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        @TaskAction
+        open fun runTest() {
+            val files = testExecutables.filter { it.exists() }.files.toTypedArray()
+            if (files.isNotEmpty()) {
+                project.exec {
+                    commandLine("xcrun", "simctl", "spawn", "iPhone X", *files)
+                }
+            } else {
+                logger.error("No test executable for iOS")
```
Comment (lukaville) [inline buildSrc/src/main/kotlin/IosPlugin.kt]: maybe we should throw an exception if it is impossible to run tests?

Comment (CherryPerry) [reply buildSrc/src/main/kotlin/IosPlugin.kt]: We have modules without tests and I don't know a way to verify if test binary will be created or not. As I know, generate test binary task is still executable even there is no tests. It just generates nothing as output.

Comment (lukaville) [reply buildSrc/src/main/kotlin/IosPlugin.kt]: :(

giphy

**Pull request merged**